internal/testrunner/testrunner.go

```diff
@@ -141,26 +148,34 @@ func FindTestFolders(packageRootPath string, testType TestType, dataStreams []st
 }
 
 // RegisterRunner method registers the test runner.
-func RegisterRunner(testType TestType, runFunc RunFunc) {
-	runners[testType] = runFunc
+func RegisterRunner(runner TestRunner) {
+	runners[runner.Type()] = runner
 }
 
 // Run method delegates execution to the registered test runner, based on the test type.
 func Run(testType TestType, options TestOptions) ([]TestResult, error) {
-	runFunc, defined := runners[testType]
+	runner, defined := runners[testType]
 	if !defined {
 		return nil, fmt.Errorf("unregistered runner test: %s", testType)
 	}
-	return runFunc(options)
+	return runner.Run(options)
 }
 
-// TestTypes method returns registered test types.
-func TestTypes() []TestType {
-	var testTypes []TestType
-	for t := range runners {
-		testTypes = append(testTypes, t)
+// TestRunners method returns registered test runners.
+func TestRunners() map[TestType]TestRunner {
+	return runners
+}
+
+// NoConfigTestRunners returns test runners that don't need configuration.
+func NoConfigTestRunners() []TestRunner {
```

Comment:
I admit I got lost here, why some test runners are defined differently (as no-config runners). I assume we can iterate on this in follow-ups.

Reply:
Yes, this was something I was experimenting with early on in this PR but is no longer necessary. I've removed it in 0e527da. Thanks for flagging it!